crates/store/re_redap_client/src/grpc.rs

```diff
@@ -590,6 +574,33 @@ async fn stream_partition_from_server(
         return Ok(());
     }
 
+    if let Some(on_ui_cmd) = on_ui_cmd {
+        // These commands only make sense to execute, once the store info has been set as they all reference
+        // the store id.
+        let store_id = store_id.clone();
+        if tx
+            .execute_on_receive(move || {
+                if let Some(time_range) = time_range {
+                    on_ui_cmd(UiCommand::SetLoopSelection {
+                        recording_id: store_id.clone(),
+                        timeline: time_range.timeline,
+                        time_range: time_range.into(),
+                    });
+                }
+                if !fragment.is_empty() {
+                    on_ui_cmd(UiCommand::SetUrlFragment {
+                        recording_id: store_id.clone(),
+                        fragment,
+                    });
+                }
+            })
+            .is_err()
+        {
+            re_log::debug!("Receiver disconnected");
+            return Ok(());
+        }
+    }
```

Comment:
Ooof, I don't like this at all. This is a callback calling a callback. Trying to figure out why the loop selection changed in the middle of receiving messages will not be fun.

I would much prefer that we converted this into data:


```rs
tx.send(LogMsg::UiCommand(UiCommand::SetLoopSelection { … }));
tx.send(LogMsg::UiCommand(UiCommand::SetUrlFragment { … }));
```

Reply:
Not saying my solution is pretty, but I'd like to avoid touching `LogMsg` though if possible. We wanted to remove it for a while and that makes it harder

Reply:
trying to find a different solution....